### PR TITLE
[mpg123] add Windows ARM support

### PR DESCRIPTION
**Pull request by driver1998**

### CODE REVIEW

File: ports/mpg123/portfile.cmake [hunk 1]
```diff
@@ -6,8 +6,16 @@ set(MPG123_HASH f226317dddb07841a13753603fa13c0a867605a5a051626cb30d45cfba266d3d
 #architecture detection
 if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
    set(MPG123_ARCH Win32)
+   set(MPG123_CONFIGURATION _x86)
 elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
    set(MPG123_ARCH x64)
+   set(MPG123_CONFIGURATION _x86)
```
Comment (vicroms) [inline ports/mpg123/portfile.cmake]: Is `set(MPG123_CONFIGURATION _X86)` correct even for **x64**?

File: ports/mpg123/portfile.cmake [hunk 2]
```diff
@@ -29,6 +37,7 @@ vcpkg_extract_source_archive_ex(
     PATCHES
         0001-fix-crt-linking.patch
         0002-fix-x86-build.patch
+        0003-add-arm-configs.patch
```
Comment (vicroms) [inline ports/mpg123/portfile.cmake]: This is quite a large patch, are there any plans to upstream these changes?

File: ports/mpg123/portfile.cmake (hunk 1 above)
Comment (driver1998) [reply ports/mpg123/portfile.cmake]: Yes, x86 here means it uses x86 ASM.

**Comment by vicroms** [comment]
Thanks for the PR!

**Pull request merged**